homeassistant/components/efergy/sensor.py

```diff
@@ -65,99 +99,99 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     dev = []
     for variable in config[CONF_MONITORED_VARIABLES]:
-        if variable[CONF_TYPE] == CONF_CURRENT_VALUES:
-            url_string = f"{_RESOURCE}getCurrentValuesSummary?token={app_token}"
-            response = requests.get(url_string, timeout=10)
-            for sensor in response.json():
-                sid = sensor["sid"]
+        for description in SENSOR_TYPES:
+            if variable[CONF_TYPE] == description.key:
+                if (
+                    description.key == CONF_CURRENT_VALUES
+                    and description.key in config[CONF_MONITORED_VARIABLES]
+                ):
+                    url_string = f"{_RESOURCE}getCurrentValuesSummary?token={app_token}"
+                    response = requests.get(url_string, timeout=10)
+                    for sensor in response.json():
+                        sid = sensor["sid"]
+                        dev.append(
+                            EfergySensor(
+                                app_token,
+                                utc_offset,
+                                variable[CONF_PERIOD],
+                                variable[CONF_CURRENCY],
+                                description,
+                                sid,
+                            )
+                        )
                 dev.append(
                     EfergySensor(
-                        variable[CONF_TYPE],
                         app_token,
                         utc_offset,
                         variable[CONF_PERIOD],
                         variable[CONF_CURRENCY],
-                        sid,
+                        description,
                     )
                 )
-        dev.append(
-            EfergySensor(
-                variable[CONF_TYPE],
-                app_token,
-                utc_offset,
-                variable[CONF_PERIOD],
-                variable[CONF_CURRENCY],
-            )
-        )
 
     add_entities(dev, True)
 
 
 class EfergySensor(SensorEntity):
     """Implementation of an Efergy sensor."""
 
-    def __init__(self, sensor_type, app_token, utc_offset, period, currency, sid=None):
+    def __init__(
+        self,
+        app_token,
+        utc_offset,
+        period,
+        currency,
+        description: SensorEntityDescription,
+        sid=None,
+    ):
         """Initialize the sensor."""
         self.sid = sid
```

Comment:
Store the entity desription
```suggestion
        self.sid = sid
        self.entity_description = description
```